cmd/cluster-agent/app/app.go

```diff
@@ -3,7 +3,7 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-2020 Datadog, Inc.
 
-// +build kubeapiserver
+// +build kubeapiserver cloudfoundry
```

Comment:
Is it a requirement to build 2 different flavour of the "cluster-agent" ?

I was thinking more to switch from "kubernetes" to "cloudfoundry" thanks to a configuration or parameter. 
I feel like it will be less work for releasing (increase the number of release job, additional docker image to publish...). 
Also, this will encourage us to define properly the internal components, their interfaces and interactions.

Reply:
It's not a strict requirement, but there are some advantages to this:

* The size of the kubernetes built binary won't be affected by adding cloudfoundry dependencies and vice versa.
* For example, right now the hostname provider for kubernetes always runs even if I try to not run the cluster agent on kubernetes (resulting in error lines in logs, as it can't find the kubeapiserver); this is due to the fact that hostname providers don't have access to any kind of global context to know where the agent is being run. I'm not familiar enough with the overall architecture to know whether this might be the case in more places or how to better address this.

As for publishing artifacts, we'll need a Bosh release (like we have for standard datadog-agent [1]), not a Docker image anyway.

Either way, I'm definitely open to suggestions. I think my approach should work quite well even without additional build tags (with some minor modifications).

[1] https://github.com/DataDog/datadog-agent-boshrelease/